From b329a14adc40868965076a77210452e396243dcd Mon Sep 17 00:00:00 2001
From: nitriques <nico@nitriques.com>
Date: Mon, 23 May 2016 14:09:34 -0400
Subject: [PATCH] Better php.ini overrides for insecure setups

This change is to protect our users against a poorly setup server. PHP
can allow pretty scary things security-wise, so it's best to make sure
things that can only have one valid setting should be enforced.

Thanks to @hyp3rlinx for this.
---
 symphony/lib/core/class.session.php | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/symphony/lib/core/class.session.php b/symphony/lib/core/class.session.php
index c0075a166..dedb526fa 100644
--- a/symphony/lib/core/class.session.php
+++ b/symphony/lib/core/class.session.php
@@ -58,6 +58,9 @@ public static function start($lifetime = 0, $path = '/', $domain = null, $httpOn
 
             if (session_id() == '') {
                 ini_set('session.save_handler', 'user');
+                ini_set('session.use_trans_sid', '0');
+                ini_set('session.use_strict_mode', '1');
+                ini_set('session.use_only_cookies', '1');
                 ini_set('session.gc_maxlifetime', $lifetime);
                 ini_set('session.gc_probability', '1');
                 ini_set('session.gc_divisor', Symphony::Configuration()->get('session_gc_divisor', 'symphony'));
